src/services/userService.js

```diff
@@ -48,28 +48,27 @@ async function getById(id) {
   return User.find(id);
 }
 
+//updateUser can be invoked by adminController and userController
+//accept id of user which will be updated, new data and isAdmin property
+//isAdmin property can change yser role and delte value of user record
 async function updateUser(id, requestBody, isAdmin) {
   const existingUser = await User.findById(id);
-  /*TODO
-  if there is empty string in req.body- mongo
-  will kept the old record. 
-  Need one more check if field must be cleared
-  */
+  if (!existingUser) {
+    throw new Error('User not found');
+  }
+
   //TODO - check functionality with liked events
-  existingUser.email = requestBody.email
-    ? requestBody.email
-    : existingUser.email;
-  existingUser.firstName = requestBody.firstName
-    ? requestBody.firstName
-    : existingUser.firstName;
-  existingUser.lastName = requestBody.lastName
-    ? requestBody.lastName
-    : existingUser.lastName;
-  existingUser.region = requestBody.region
-    ? requestBody.region
-    : existingUser.region;
+  for (let key of Object.keys(requestBody)) {
+    if (key == 'email' && requestBody[key] == '') {
```

Comment:
Same here with isDeleted

Reply:
isAdmin cames after parsing token
isDeleted will be added